addon/adapters/rest.js

```diff
@@ -1154,6 +1154,10 @@ var RESTAdapter = Adapter.extend(BuildURLMixin, {
   buildQuery(snapshot) {
     let query = {};
 
+    if (snapshot) {
+      query.adapterOptions = snapshot.adapterOptions
+    }
```

Comment:
This isn't needed, the main thing this PR tries to solve is just passing the `snapshot` to the `urlForFindBelongsTo` and `urlForFindHasMany` hook. Sorry if my comments confused you here...